lib/classes/hint.dart

```diff
@@ -0,0 +1,41 @@
+import 'dart:math';
+
+import 'package:phone_factory/classes/Ship.dart';
+import 'package:phone_factory/classes/bay_item.dart';
+
+class Hint {
+  int _currentBayNumber;
+  Ship ship = Ship();
+
+  Hint({required int currentBayNumber}) : _currentBayNumber = currentBayNumber;
+
+  void getHint() {
+    Map<int, dynamic> baysData = ship.getSpaceShipBays();
+    List<int> alloweBaysList = List.from(baysData.keys);
+    alloweBaysList.remove(_currentBayNumber);
+    Random random = Random();
+    int randomBayIndex = random.nextInt(alloweBaysList.length);
+    int randomAllowedBay = alloweBaysList[randomBayIndex];
+    BayItem randomBay = baysData[randomAllowedBay];
```

Comment:
no need to create variable for this you can write switch (baysData[randomAllowedBay])